File: Definitions/Game_Definitions/classes/SoundSystem.lua
---@meta
---@diagnostic disable

---@class SoundSystem : InteractiveMasterDevice
SoundSystem = {}

---@return SoundSystem
function SoundSystem.new() return end

---@param props table
---@return SoundSystem
function SoundSystem.new(props) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function SoundSystem:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function SoundSystem:OnTakeControl(ri) return end

---@return SoundSystemController
function SoundSystem:GetController() return end

---@return SoundSystemControllerPS
function SoundSystem:GetDevicePS() return end

File: DecompiledGameScripts/cyberpunk/devices/masters/soundSystem/soundSystem.swift

public class SoundSystem extends InteractiveMasterDevice {

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as SoundSystemController;
  }

  private const func GetController() -> ref<SoundSystemController> {
    return this.m_controller as SoundSystemController;
  }

  public const func GetDevicePS() -> ref<SoundSystemControllerPS> {
    return this.GetController().GetPS();
  }
}
